modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2259 @@
+        params.set<MaterialPropertyName>(NS::mu) = viscosity_name;
+
+        for (unsigned int d = 0; d < _dim; ++d)
+        {
+          NonlinearVariableName vname;
+          if (_porous_medium_treatment)
+          {
+            vname = NS::superficial_velocity_vector[d];
+            for (unsigned int d = 0; d < _dim; ++d)
+              params.set<MooseFunctorName>(u_names[d]) = NS::superficial_velocity_vector[d];
+            params.set<UserObjectName>("rhie_chow_user_object") = "pins_rhie_chow_interpolator";
+          }
+          else
+          {
+            vname = NS::velocity_vector[d];
+            for (unsigned int d = 0; d < _dim; ++d)
+              params.set<MooseFunctorName>(u_names[d]) = NS::velocity_vector[d];
+            params.set<UserObjectName>("rhie_chow_user_object") = "ins_rhie_chow_interpolator";
+          }
+
+          params.set<NonlinearVariableName>("variable") = vname;
+          params.set<MooseEnum>("momentum_component") = NS::directions[d];
+
+          _problem->addFVBC(bc_type, vname + "_" + _wall_boundaries[bc_ind], params);
+        }
+      }
+      {
+        const std::string bc_type = "INSFVSymmetryPressureBC";
+        InputParameters params = _factory.getValidParams(bc_type);
+        params.set<NonlinearVariableName>("variable") = NS::pressure;
+        params.set<std::vector<BoundaryName>>("boundary") = {_wall_boundaries[bc_ind]};
+
+        _problem->addFVBC(bc_type, NS::pressure + "_" + _wall_boundaries[bc_ind], params);
+      }
+    }
+  }
+}
+
+void
+NSFVAction::addINSEnergyWallBC()
+{
+  unsigned int skipped_energy_bc = 0;
+
+  for (unsigned int bc_ind = 0; bc_ind < _wall_boundaries.size(); ++bc_ind)
+  {
+    if (_energy_wall_types[bc_ind] == "fixed-temperature")
+    {
+      const std::string bc_type = "FVFunctionDirichletBC";
+      InputParameters params = _factory.getValidParams(bc_type);
+      params.set<NonlinearVariableName>("variable") = NS::T_fluid;
+      params.set<FunctionName>("function") = _energy_wall_function[bc_ind];
+      params.set<std::vector<BoundaryName>>("boundary") = {_wall_boundaries[bc_ind]};
+
+      _problem->addFVBC(bc_type, NS::T_fluid + "_" + _wall_boundaries[bc_ind], params);
+    }
+    else if (_energy_wall_types[bc_ind] == "heatflux" || _energy_wall_types[bc_ind] == "symmetry")
+    {
+      const std::string bc_type = "FVFunctionNeumannBC";
+      InputParameters params = _factory.getValidParams(bc_type);
+      params.set<NonlinearVariableName>("variable") = NS::T_fluid;
+      if (_energy_wall_types[bc_ind] == "symmetry")
+      {
+        params.set<FunctionName>("function") = "0.0";
+        skipped_energy_bc += 1;
+      }
+      else
+        params.set<FunctionName>("function") = _energy_wall_function[bc_ind - skipped_energy_bc];
+
+      params.set<std::vector<BoundaryName>>("boundary") = {_wall_boundaries[bc_ind]};
+
+      _problem->addFVBC(bc_type, NS::T_fluid + "_" + _wall_boundaries[bc_ind], params);
+    }
```

Comment:
techincally you would want the energy flux BCs here too.
BUT with Alex's changes we wont need to be using flux BCs, and will be able to have conservation with Dirichlet.
So I'd say no need to bother